### DISCUSSION TITLE
Focus on Custom Tests, Not Standard ERC20

### DISCUSSION THREAD

**Discussion by spencerj411** [Cyfrin/foundry-full-course-cu#2479] [General]
Hi Cyfrin Updraft team!

I've just finished watching the [Test your ERC20 using AI](https://updraft.cyfrin.io/courses/advanced-foundry/How-to-create-an-erc20-crypto-currency/erc20-ai-tests-and-recap) lesson. While I understand that the purpose of this lesson is to introduce the basics of ERC20 and to get familiar with its different functions, I wanted to point out that when developing your own ERC20 token, it's generally unnecessary to write unit tests that re-test core ERC20 functionality. These core features are already well-tested and community-vetted, especially when using OpenZeppelin contracts.

Instead, when writing unit tests, it's best to focus on testing the additional functionality you build on top of the ERC20 standard.

Please let me know down below your thoughts. Thanks, team!

**Comment by EngrPips** [comment]
You have a point!
